cli/plasmo/src/features/manifest-factory/base.ts

```diff
@@ -409,7 +409,8 @@ export abstract class PlasmoManifest<T extends ExtensionManifest = any> {
 
   addDirectory = async (
     path: string,
-    toggleDynamicPath: typeof this.toggleContentScript
+    toggleDynamicPath: typeof this.toggleContentScript,
+    filterFile?: (fileName: string) => boolean
```

Comment:
I originally had a separate `addIndexDirectory` method in [aee64a9](https://github.com/PlasmoHQ/plasmo/pull/657/commits/aee64a9da056541656c0213edef912c6aa9f7002) which was nearly identical with the addition of filtering out non-index files. This felt a bit redundant, so I opted to add an optional filter function arg - open to alternate approaches!